src/DetectSecretsLinter.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Copyright 2020 Pinterest, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Detect secrets in code base. 
+ */
+final class DetectSecretsLinter extends ArcanistExternalLinter {
+
+  public function getInfoName() {
+      return 'detect-secrets'; 
+  }
+
+  public function getInfoURI() {
+      return 'https://github.com/Yelp/detect-secrets';
+  }
+
+  public function getInfoDescription() {
+      return 'Detect secrets within code base during compile time';
```

Comment:
I don't think "during compile time" is quite right here given that this is a lint step.